### ISSUE TITLE
Doesn't work on Android 14

### ISSUE THREAD

**Issue by tyrell-wallace** [jiro-aqua/aGrep#11]
It used to work just fine on my Android 13, but then I updated to Android 14 and it doesn't work anymore.

**Comment by ajburley** [comment]
Yes it's because of the new target SDK minimum which Google introduced on Android 14. Apps must target Android 6 or above, this one is still targeting Android 5.0. Unfortunately the app hasn't been updated for over 10 years; I have emailed the author just in case I can get a response, otherwise may just have to fork it.

**Comment by kappa** [comment]
I burned a bunch of AI tokens to get it to working state here: https://github.com/kappa/aGrep/releases/tag/v1.0.0

**Comment by ajburley** [comment]
@kappa are you planning to publish this on Google Play? I also started working on my own fork, but I haven't made as much progress as you yet.

**Comment by jpberry** [comment]
Hi, I tried to install kappa's code, but I get a "App not installed as package appears to be invalid".
But thanks for the help : I've been using agrep a lot, until I switched phones...

**Comment by kappa** [comment]
@ajburley I would love to, yes. Cannot commit to any dates though.

**Comment by kappa** [comment]
@jpberry I think I messed up with signing, I've never fully understood how that works with automatic builds. Can you try this one? https://media.freefeed.net/attachments/60eabc59-da07-48d9-93a9-85d284ced831.apk

**Comment by jpberry** [comment]
works fine. Many, many thanks, my home screen looked f**ed up without it...

**Comment by kappa** [comment]
https://play.google.com/store/apps/details?id=net.kappa.agrep_again&hl=en-us

**Comment by ajburley** [comment]
> https://play.google.com/store/apps/details?id=net.kappa.agrep_again&hl=en-us

@kappa Thank you! It works, but seems **really** slow compared to the original aGrep or even toybox grep. To give an example, a search for a string in ~8000 files took around 8 seconds in toybox grep, but took 164 seconds in aGrep Again (installed from Play Store). I guess this is maybe just because SAF URIs are inherently slow, but it makes the speed rather prohibitive for me (having to wait almost 3 minutes to do something which I could do in 8 seconds). I don't know if a version of the app could use all-files access rather than SAF, either in a variant not published on Google Play, or even on Google Play itself, as "on-device file search" is given by Google as a use case apps can use to justify needing all files access.

**Comment by kappa** [comment]
> > https://play.google.com/store/apps/details?id=net.kappa.agrep_again&hl=en-us
> 
> [@kappa](https://github.com/kappa) Thank you! It works, but seems **really** slow compared to the original aGrep or even toybox grep. To give an example, a search for a string in ~8000 files took around 8 seconds in toybox grep, but took 164 seconds in aGrep Again (installed from Play Store). I guess this is maybe just because SAF URIs are inherently slow, but it makes the speed rather prohibitive for me (having to wait almost 3 minutes to do something which I could do in 8 seconds). I don't know if a version of the app could use all-files access rather than SAF, either in a variant not published on Google Play, or even on Google Play itself, as "on-device file search" is given by Google as a use case apps can use to justify needing all files access.

I fully agre, it is really slow. I have an idea that will require the user to allow aGrep full access to the storage in the settings. Google will have to allow that special permission but I think it will be much easier to get that permission if I show that using SAF for grep is just too slow.

**Comment by ajburley** [comment]
@kappa yeah that's exactly what I meant by "use all-files access rather than SAF"...and as I said, Google should allow it as "on-device file search" is one of their explicitly listed use cases for that permission. Good luck, and I look forward to trying it out!
